### DISCUSSION TITLE
Result/Report

### DISCUSSION THREAD

**Discussion by souravsk** [kubescape/kubescape#792] [Q&A] [answered]
hi @vladklokun 
I was trying to Print the result/report in markdown format. So while going through your code for the HTML format. I didn't know how to execute htmlprinter.go file. So can you let me know the commands to execute those codes?
As per my knowledge, HTML and markdown are somewhat simpler so I was also using your code to create the markdown file.
Or I can also use our .Html file and then convert it into markdown by using python if that is possible

**Comment by vladklokun** [comment, marked as answer]
Hi!

The `htmlprinter.go` file is not executable by itself. It contains functions that are used elsewhere to generate a Kubescape report in the HTML format.

As far as I understand, you are trying to receive a Kubescape report in Markdown format. Is that right? If so, then I′m afraid that at the moment, Kubescape does not support generating Markdown reports.

If you desperately need a Markdown report, then you should generate a report in one of the formats we support—be it HTML, JUnit or something else—and then try to convert it to Markdown. You might have some luck with the HTML format, since we have tried to markup our data as semantically as possible.

Could you please explain _why_ do you want a Kubescape report in Markdown format?

**Reply by souravsk** [reply]
NO that's not what I meant. what I'm trying to say is that when I looked at the roadmap in the planning session it was mention that it you can add the result in markdown so i was working on that feature. 
My question is 
- what is the command to get the result in HTML 
- Is any thing where you can help me making the markdown.go file

**Reply by souravsk** [reply]
@vladklokun

**Reply by souravsk** [reply]
i need the command to run the HTML result

**Reply by dwertent** [reply]
HTML format output [example](https://github.com/kubescape/kubescape#output-in-html-format)

**Reply by souravsk** [reply]
thanks @dwertent sorry I didn't see it was already available in the readme file

**Reply by vladklokun** [reply]
> I looked at the roadmap in the planning session

@souravsk Could you link this roadmap, please? Or explain what is this planning session that you’ve mentioned?

Because I am not aware of any plans to add support for generating reports in the Markdown format, it’s not mentioned in our [Roadmap](https://github.com/kubescape/kubescape/blob/eb16440ba646e64ff6123ef4bcaca981d084c7c5/docs/roadmap.md) and I don’t want to waste anyone’s time on it.

**Reply by souravsk** [reply]
it was mentioned previously so I started working on it after some time checking it as removed from the roadmap. so asked in the discord  some one said you can work on that we still don't support for markdown format

**Comment by souravsk** [comment]
Hi @vladklokun can you let me know how did you link this file https://github.com/souravsk/kubescape/blob/master/core/pkg/resultshandling/printer/v2/html/report.gohtml
to the htmlprinter.go . i'm trying to understand how the file is get connected which the go file because there is no mention of the file. why the name of the file is .gohtml by not HTML

**Reply by souravsk** [reply]
if anyone other than @vladklokun knows about this? please help me to understand

**Reply by vladklokun** [reply]
I have embedded the contents of the `report.gohtml` file into a variable with a `go:embed` directive [here](https://github.com/souravsk/kubescape/blob/6e2dda799336bd857a7c68ee86f9815f4a3d5a42/core/pkg/resultshandling/printer/v2/htmlprinter.go#L25).

**Reply by souravsk** [reply]
but it's commanded shouldn't GO just ignore that line.

**Reply by vladklokun** [reply]
As with many [Go directives](https://pkg.go.dev/cmd/compile#hdr-Compiler_Directives) like `go:build` and `go:generate`, it’s a special case, as documented in the [`embed` package](https://pkg.go.dev/embed) and [Go by Example](https://gobyexample.com/embed-directive).

**Reply by souravsk** [reply]
okay, thank you so much for the replay. I'm trying to replicate your work to get the output in markdown format.
Anything you like to share otherwise thank you for help

**Reply by vladklokun** [reply]
Before you do any work, let’s first figure out what is this roadmap in the planning session that you’ve mentioned. Because I’m not sure that we want this feature. Please take a look at [this comment](https://github.com/kubescape/kubescape/discussions/792#discussioncomment-3682923).
